granary/source.py

```diff
@@ -801,7 +801,23 @@ def changed(b, a, field, label):
     return any(changed(before, after, field, 'activity') or
                changed(obj_b, obj_a, field, 'activity[object]')
                for field in ('objectType', 'verb', 'to', 'content', 'location',
-                             'image'))
+                             'image', 'inReplyTo'))
+
+  @staticmethod
+  def append_in_reply_to(before, after):
+    """appends the inReplyTos from the before object to the after object, in place
+
+    Args:
+      before, after: dicts, ActivityStreams activities or objects
+    """
+    obj_b = before.get('object', before)
+    obj_a = after.get('object', after)
+
+    if obj_b and obj_a:
+      reply_b = util.get_list(obj_b,'inReplyTo')
+      reply_a = util.get_list(obj_a,'inReplyTo')
+      if reply_b and reply_a:
```

Comment:
minor nit, you can skip this, since `get_list` intentionally returns the empty list if the field is missing, to make code like this easier.